kong/runloop/plugins_iterator.lua

```diff
@@ -321,14 +321,28 @@ function PluginsIterator.new(version)
     end
 
     if should_process_plugin(plugin) then
-      map[plugin.name] = true
+      local name = plugin.name
+
+      map[name] = true
 
       local combo_key = (plugin.route    and 1 or 0)
                       + (plugin.service  and 2 or 0)
                       + (plugin.consumer and 4 or 0)
 
-      combos[plugin.name] = combos[plugin.name] or {}
-      combos[plugin.name][combo_key] = true
+      combos[name]          = combos[name]          or {}
+      combos[name].both     = combos[name].both     or {}
+      combos[name].routes   = combos[name].routes   or {}
+      combos[name].services = combos[name].services or {}
+
+      combos[name][combo_key] = true
+
+      if plugin.route and plugin.service then
+        combos[name].both[plugin.route.id] = plugin.service.id
+      elseif plugin.route then
+        combos[name].routes[plugin.route.id] = true
+      elseif plugin.service then
```

Comment:
style: ideally we should be adding line breaks before the `elseif` branches, as usual.

Reply:
line breaks added.